tsdb/head_test.go

```diff
@@ -5819,3 +5819,16 @@ func TestHeadAppender_AppendCTZeroSample(t *testing.T) {
 		require.Equal(t, chunkenc.ValNone, it.Next())
 	}
 }
+
+func TestHeadCompactableDoesNotRequireOverflow(t *testing.T) {
+	// Set a chunk range of 1 to ensure that Head.compactable() doesn't require
```

Comment:
Nice catch!
I'll be picky about the comment, IIUC, on `main` this test fails, so the overflow doesn't help at all (quite the opposite). The comment implies it does and we wanted to make it explicit. If we can adjust that it'd be great.

---

I think we can have this problem even with initialized heads, for the `h.chunkRange.Load()=1` ones: if the diff between `h.MaxTime()` and `h.MinTime()` is `1`, the head will be compactable, because we divide before multiply in `h.chunkRange.Load()/2*3` we lose precision, maybe we should change the order.

Reply:
maybe the second part can be discussed/done in another PR.

Reply:
Update the comment and test name to more accurately describe what we're testing.

Reply:
Thanks.